src/renderer/opengl.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+#include "../math/types.h"
+
+namespace GL {
+
+constexpr u32 cARRAY_BUFFER = 0x8892;
+constexpr u32 cBLEND = 0x0BE2;
+constexpr u32 cCOMPILE_STATUS = 0x8B81;
+constexpr u32 cDEBUG_OUTPUT = 0x92E0;
+constexpr u32 cDEBUG_TYPE_ERROR = 0x824C;
+constexpr u32 cDEPTH_BUFFER_BIT = 0x00000100;
+constexpr u32 cDEPTH_TEST = 0x0B71;
+constexpr u32 cCOLOR_BUFFER_BIT = 0x00004000;
+constexpr u32 cFLOAT = 0x1406;
+constexpr u32 cFRAGMENT_SHADER = 0x8B30;
+constexpr u32 cFRAMEBUFFER = 0x8D40;
+constexpr u32 cFRAMEBUFFER_COMPLETE = 0x8CD5;
+constexpr u32 cINT = 0x1404;
+constexpr u32 cLINK_STATUS = 0x8B82;
+constexpr u32 cRED = 0x1903;
+constexpr u32 cRENDERBUFFER = 0x8D41;
+constexpr u32 cRG = 0x8227;
+constexpr u32 cRGB = 0x1907;
+constexpr u32 cRGBA = 0x1908;
+constexpr u32 cSRC_ALPHA = 0x0302;
+constexpr u32 cSTATIC_DRAW = 0x88E4;
+constexpr u32 cSTREAM_DRAW = 0x88E0;
+constexpr u32 cTEXTURE0 = 0x84C0;
+constexpr u32 cTEXTURE1 = 0x84C1;
+constexpr u32 cTEXTURE_2D = 0x0DE1;
+constexpr u32 cTEXTURE_2D_ARRAY = 0x8C1A;
+constexpr u32 cTRIANGLES = 0x0004;
+constexpr u32 cUNIFORM_BUFFER = 0x8A11;
+constexpr u32 cUNSIGNED_BYTE = 0x1401;
+constexpr u32 cVERTEX_SHADER = 0x8B31;
+
+#ifdef _SMEK_GL_IMPLEMENT
+#define GL_FUNC(ret, name, ...)\
+    typedef ret (* _ ## name ## _func)(__VA_ARGS__);\
+    _ ## name ## _func name = nullptr;
+#else
+#define GL_FUNC(ret, name, ...)\
+    typedef ret (* _ ## name ## _func)(__VA_ARGS__);\
+    extern _ ## name ## _func name;
+#endif
+
```

Comment:
This flag should probably be documented somewhere.

Reply:
No one should use it, it's only for making sure the implementations end up in one compilation unit, and one compilation unit only. So... I don't think so... See opengl.cpp:5

Reply:
:+1: